api/rql/find.go

```diff
@@ -0,0 +1,17 @@
+package rql
+
+import (
+	"context"
+
+	"github.com/puppetlabs/wash/plugin"
+)
+
+// Find returns all children of the start entry that satisfy the given query
+// (including the start entry itself). Note that all returned entries' paths
+// will start from "", which represents the start path. For example, if "childOne"
+// and "childTwo" are the cnames of the start entry's children, then their "Path"
+// fields will be set to "childOne" and "childTwo" (where the start entry's path of
+// "" is automatically prefixed).
+func Find(ctx context.Context, start plugin.Entry, query Query, options Options) ([]Entry, error) {
```

Comment:
Thinking the RQL should exclude the start path given that `Path` and `Kind` don't match it. Then the spec is "this returns all descendants of path that satisfy the given query"

Reply:
Also one annoying this is we can't enforce order of returned values. We do know that returned values will always be parent then children, but there's no ordering in which parent is traversed. E.g., given something like `.`, `./foo`, `./foo/1`, `./bar`, we can always assume that `./foo` will come before `./foo/1` in the return results, but we cannot ordering of `./bar` and `./foo`.

This is b/c of `plugin.EntryMap`, which stores things in a `map` object and `map` has no order. Maybe this is OK?

Reply:
We usually sort them before traversing when it matters.

Reply:
Ok, updated to sort properly.

What about excluding the start path?

Reply:
Not sure I understand the impact yet.

Reply:
Yeah, I'd exclude start path. It feels clearer, and consistent with what you'd expect (how `find` currently works).

Reply:
Oh, I was wrong about how `find` works. But it would be difficult to implement the way `find` works in the API, so it feels better to exclude start path in the API and if we want our `find` command to work consistent with the BSD implementation then we can handle that in the command.

Reply:
Updated to exclude the start path.